src/ck-pics/trace-perf.C

```diff
@@ -328,43 +341,54 @@ void TraceAutoPerf::summarizeObjectInfo(double &maxtime, double &totaltime,
       maxMsgSize = it->second->msgSize;
     totalMsgSize += it->second->msgSize;
     numObjs++;
+
+    double temp = (double)(it->second->msgSize)/(double)(it->second->msgCount);
+    if(temp > maxBytesPerMsg) {
+      maxBytesPerMsg = temp;
+    }
+    if(temp < minBytesPerMsg) {
+      minBytesPerMsg = temp;
+    }
   }
   numObjs += numNewObjects;
 }
 
 PerfData* TraceAutoPerf::getSummary() {
   if(isTraceOn){
-  currentSummary->data[AVG_TotalTime] = CkWallTimer()-startTimer;
-  currentSummary->data[AVG_IdlePercentage] = currentSummary->data[MIN_IdlePercentage]= currentSummary->data[MAX_IdlePercentage]= (idleTime())/currentSummary->data[AVG_TotalTime]; 
-  currentSummary->data[MAX_LoadPerPE] = currentSummary->data[AVG_TotalTime] - idleTime();
-  currentSummary->data[MIN_UtilizationPercentage] = currentSummary->data[MAX_UtilizationPercentage] = (utilTime())/currentSummary->data[AVG_TotalTime]; 
-  currentSummary->data[AVG_UtilizationPercentage] = utilTime()/currentSummary->data[AVG_TotalTime];
-  currentSummary->data[MIN_AppPercentage] = currentSummary->data[MAX_AppPercentage] = appTime();
-  currentSummary->data[AVG_AppPercentage] = appTime();
-  currentSummary->data[AVG_TuningOverhead] = tuneOverheadTotalTime; 
-  currentSummary->data[MIN_OverheadPercentage] = currentSummary->data[MAX_OverheadPercentage] = overheadTime(); 
-  currentSummary->data[AVG_OverheadPercentage] = overheadTime()/currentSummary->data[AVG_TotalTime];
+  currentSummary->data[AVG_TotalTime] = totalTraceTime();
+  currentSummary->data[MIN_IdlePercentage] = currentSummary->data[MAX_IdlePercentage] = currentSummary->data[AVG_IdlePercentage] = idleRatio();
+  currentSummary->data[MIN_OverheadPercentage] = currentSummary->data[MAX_OverheadPercentage] = currentSummary->data[AVG_OverheadPercentage] = overheadRatio();
+  currentSummary->data[MIN_UtilizationPercentage] = currentSummary->data[MAX_UtilizationPercentage] = currentSummary->data[AVG_UtilizationPercentage] = utilRatio();
+  currentSummary->data[MAX_NumInvocations] = currentSummary->data[AVG_NumInvocations] = currentSummary->data[AVG_NumInvocations] = (double)totalEntryMethodInvocations;
+
+  summarizeObjectInfo(currentSummary->data[MAX_LoadPerObject], currentSummary->data[AVG_LoadPerObject], currentSummary->data[MIN_LoadPerObject],
+                      currentSummary->data[MAX_NumMsgsPerObject],  currentSummary->data[AVG_NumMsgsPerObject],
+                      currentSummary->data[MAX_BytesPerObject], currentSummary->data[AVG_BytesPerObject], currentSummary->data[AVG_NumObjectsPerPE],
+                      currentSummary->data[MAX_BytesPerMsg], currentSummary->data[MIN_BytesPerMsg]);
+  currentSummary->data[AVG_NumMsgRecvPerPE] = currentSummary->data[AVG_NumMsgsPerObject];
+  currentSummary->data[AVG_BytesMsgRecvPerPE] = currentSummary->data[AVG_BytesPerObject];
```

Comment:
Why is this using the per object values to set the per PE values? Are they combined somehow somewhere else?

Reply:
My bad. This is an example of using incorrect variable names. currentSummary->data[AVG_NumMsgsPerObject] is the AVG_NumMsgRecvPerPE, which is later converted to Avg_NumMsgPerObject. Same for the second one. Also leads to the comments you mentioned https://github.com/UIUC-PPL/charm/pull/3419#discussion_r674142593 here.
Will fix this in the next commit.